apps/veyfi/components/LockTab.tsx

```diff
@@ -0,0 +1,174 @@
+import {useEffect, useMemo, useState} from 'react';
+import {useVotingEscrow} from '@veYFI/contexts/useVotingEscrow';
+import {useTransaction} from '@veYFI/hooks/useTransaction';
+import {getVotingPower} from '@veYFI/utils';
+import * as VotingEscrowActions from '@veYFI/utils/actions/votingEscrow';
+import {MAX_LOCK_TIME, MIN_LOCK_AMOUNT, MIN_LOCK_TIME} from '@veYFI/utils/constants';
+import {validateAllowance, validateAmount, validateNetwork} from '@veYFI/utils/validations';
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {formatBN, formatUnits} from '@yearn-finance/web-lib/utils/format.bigNumber';
+import {formatAmount} from '@yearn-finance/web-lib/utils/format.number';
+import {fromWeeks, getTimeUntil, toSeconds, toTime, toWeeks} from '@yearn-finance/web-lib/utils/time';
+import {useWallet} from '@common/contexts/useWallet';
+import {useBalance} from '@common/hooks/useBalance';
+import {DefaultTNormalizedBN, toNormalizedBN} from '@common/utils';
+
+import {AmountInput} from '../../common/components/AmountInput';
+
+import type {BigNumber, ethers} from 'ethers';
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TMilliseconds} from '@yearn-finance/web-lib/utils/time';
+
+function LockTab(): ReactElement {
+	const [lockAmount, set_lockAmount] = useState(DefaultTNormalizedBN);
+	const [lockTime, set_lockTime] = useState('');
+	const {provider, address, isActive, chainID} = useWeb3();
+	const {refresh: refreshBalances} = useWallet();
+	const {votingEscrow, positions, allowances, isLoading: isLoadingVotingEscrow, refresh: refreshVotingEscrow} = useVotingEscrow();
+	const tokenBalance = useBalance(toAddress(votingEscrow?.token));
+	const clearLockAmount = (): unknown => set_lockAmount(DefaultTNormalizedBN);
+	const refreshData = (): unknown => Promise.all([refreshVotingEscrow(), refreshBalances()]);
+	const onTxSuccess = (): unknown => Promise.all([refreshData(), clearLockAmount()]);
+	const [approveLock, approveLockStatus] = useTransaction(VotingEscrowActions.approveLock, refreshData);
+	const [lock, lockStatus] = useTransaction(VotingEscrowActions.lock, onTxSuccess);
+	const [increaseLockAmount, increaseLockAmountStatus] = useTransaction(VotingEscrowActions.increaseLockAmount, onTxSuccess);
+
+	const web3Provider = provider as ethers.providers.Web3Provider;
+	const userAddress = address as TAddress;
```

Comment:
I'd suggest the same thing here as https://github.com/yearn/yearn.fi/pull/29/files#r1064538622